examples/plotting/file/custom_datetime_axis.py

```diff
@@ -0,0 +1,65 @@
+from pdb import set_trace as bp
+from math import pi
+
+import pandas as pd
+
+from bokeh.sampledata.stocks import MSFT
+from bokeh.plotting import figure, show, output_file
+from bokeh.models.formatters import TickFormatter, String, List
+
+# In this custom TickFormatter, xaxis labels are taken from an array of date
+# Strings (e.g. ['Sep 01', 'Sep 02', ...]) passed to the date_labels property. 
+class DateGapTickFormatter(TickFormatter):
+  date_labels = List(item_type=String)
+  __implementation__ = """
+_ = require "underscore"
+HasProperties = require "common/has_properties"
+
+class DateGapTickFormatter extends HasProperties
+  type: 'DateGapTickFormatter'
+
+  format: (ticks) ->
+    date_labels = @get("date_labels")
+    labels = ( 
+      for tick, i in ticks
+        date_labels[tick] ? ""
+      )
```

Comment:
As this is a one-liner, it would be better to use `(date_labels[tick] ? "" for tick in ticks)` syntax. Note also `i` is not used, so can be removed.